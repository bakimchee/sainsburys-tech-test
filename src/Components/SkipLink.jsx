import React from 'react';
import styled from 'styled-components';
import { theme } from '../Theme';

const SkipLinkStyle = styled.a`
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
  &:focus {
    clip: auto;
    height: auto;
    overflow: auto;
    position: absolute;
    width: auto;
    background-color: white;
    border: ${theme.border.width.default} blue solid;
    padding: 0.5rem;
  }
`;

const SkipLink = () => <SkipLinkStyle href="#main">Skip to main content</SkipLinkStyle>;

export default SkipLink;
