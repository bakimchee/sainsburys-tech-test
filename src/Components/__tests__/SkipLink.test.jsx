/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import SkipLink from '../SkipLink';

test('SkipLink', () => {
  const component = renderer.create(<SkipLink>Add to basket</SkipLink>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
