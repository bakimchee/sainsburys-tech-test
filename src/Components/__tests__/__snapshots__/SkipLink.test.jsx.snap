// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink 1`] = `
.c0 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0:focus {
  -webkit-clip: auto;
  clip: auto;
  height: auto;
  overflow: auto;
  position: absolute;
  width: auto;
  background-color: white;
  border: 2px blue solid;
  padding: 0.5rem;
}

<a
  className="c0"
  href="#main"
>
  Skip to main content
</a>
`;
